components/Profile/DetailUpdateModal.js

```diff
@@ -0,0 +1,118 @@
+/* eslint-disable prettier/prettier */
+//import libraries
+import React from 'react';
+import { View, Text, StyleSheet, Modal, TextInput, Alert, Button} from 'react-native';
+import color from '../../constants/color';
+
+/**
+ *
+ * @param {string} prop.toggleDetailModal function that toggles visible or invisible
+ * @param {string} prop.modalVisible Actual Boolean being toggled in parent
+ * @param {string} prop.originalDetail the original detail (username)
+ * @param {Boolean} prop.onConfirm what to do when the user clicks save
+ *
+ */
+// create a component
+const DetailUpdateModal = ({toggleDetailModal, modalVisible, originalDetail, onConfirm}) => {
+  const [detail, setDetail] = React.useState('');
+  const [errorMessage, setErrorMessage] = React.useState('');
+  const legalChars = /^[a-zA-Z0-9_-]*$/; // allow letters, numbers, and underscores
+
+  const updateUserDetails = () => {
+    if (detail.length < 5){
+      setErrorMessage('The username must be longer than 5 characters');
+      return;
+    } else if (!legalChars.test(detail)){
```

Comment:
No need of `else` here and below. Just `if` is enough

Reply:
elses removed